Python/ceval.c

```diff
@@ -5309,7 +5382,7 @@ get_exception_handler(PyCodeObject *code, int index, int *level, int *handler, i
 static int
 initialize_locals(PyThreadState *tstate, PyFrameConstructor *con,
     PyObject **localsplus, PyObject *const *args,
-    Py_ssize_t argcount, PyObject *kwnames)
+    Py_ssize_t argcount, PyObject *kwnames, int steal_args)
```

Comment:
The cleanup would be more complex, see https://github.com/python/cpython/pull/28488#discussion_r712934439

Reply:
I don't agree.
Cleanup is simpler if `initialize_locals` is consistent and always steals the references to `args` regardless of whether it succeeds or fails.
`initialize_locals` (or any other function for that matter) should leave things in an inconsistent state.

Reply:
But then what do you suggest? That `initialize_locals` always steals and `_PyEvalFramePushAndInit` is the one that increases or decreases the refcount?

Reply:
But that will require changes in all callers of `_PyEvalFramePushAndInit`, wouldn't it?

Reply:
No other function has such a contract. They either steal all the references, or they steal none of them.
Anything more complicated than that is too easy to misuse.

These functions will get more use as we extend this to `CALL_FUNCTION_KW`, `CALL_METHOD` and `CALL_METHOD_KW`, so they need an easy to use interface.

Reply:
Ok, I trust your judgement but I would like to discuss this together so we are on the same page because I still have some concerns. Let's schedule a call.

Reply:
May I add my 2 cent? I have been bitten very much with such inconsistencies, a few (20) years ago. Consistency with simplicity is a very good concept.

Reply:
I agree that long term reliability gained from consistency is absolutely worth the short term effort.

However, after a bit of experimentation, I've found that the changes required are relatively extensive, so should be done in another PR.